packages/ai/src/prompts/chat.ts

```diff
@@ -60,12 +61,36 @@ export type ChatPromptSendOptions<TOptions extends Record<string, any> = Record<
   readonly onChunk?: TextChunkHandler;
 };
 
+export type ChatPromptPlugin<TPluginName extends string, TPluginUseArgs extends {}> = {
+  /**
+   * Unique name of the plugin
+   */
+  name: TPluginName;
+
+  /**
+   * The function is set to initialize the plugin
+   * @param args - Arguments to initialize the plugin
+   */
+  usePlugin?: (args: TPluginUseArgs) => void;
+
+  /**
+   * Optionally passed in to modify the functions array that
+   * is passed to the model
+   * @param functions
+   * @returns Functions
+   */
+  onBuildFunctions?: (functions: Function[]) => Function[] | Promise<Function[]>;
+};
+
 /**
  * a prompt that can interface with a
  * chat model that provides utility like
  * streaming and function calling
  */
-export interface IChatPrompt<TOptions extends Record<string, any> = Record<string, any>> {
+export interface IChatPrompt<
+  TOptions extends Record<string, any> = Record<string, any>,
+  TChatPromptPlugins extends readonly ChatPromptPlugin<string, any>[] = [],
```

Comment:
Itd be nice to include the plugins array in the options that exist so the input looks a bit cleaner

Reply:
added plugin array, lmk if that aligns to what you were thinking